src/software/pipeline/main_triangulate.cpp

```diff
@@ -0,0 +1,275 @@
+
+    po::variables_map vm;
+    try
+    {
+        po::store(po::parse_command_line(argc, argv, allParams), vm);
+
+        if (vm.count("help") || (argc == 1))
+        {
+            ALICEVISION_COUT(allParams);
+            return EXIT_SUCCESS;
+        }
+        po::notify(vm);
+    }
+    catch (boost::program_options::required_option& e)
+    {
+        ALICEVISION_CERR("ERROR: " << e.what());
+        ALICEVISION_COUT("Usage:\n\n" << allParams);
+        return EXIT_FAILURE;
+    }
+    catch (boost::program_options::error& e)
+    {
+        ALICEVISION_CERR("ERROR: " << e.what());
+        ALICEVISION_COUT("Usage:\n\n" << allParams);
+        return EXIT_FAILURE;
+    }
+
+    ALICEVISION_COUT("Program called with the following parameters:");
+    ALICEVISION_COUT(vm);
+
+    // set verbose level
+    system::Logger::get()->setLogLevel(verboseLevel);
+
+    const double defaultLoRansacLocalizationError = 4.0;
+    if (!robustEstimation::adjustRobustEstimatorThreshold(sfmParams.localizerEstimator, sfmParams.localizerEstimatorError, defaultLoRansacLocalizationError))
+    {
+        return EXIT_FAILURE;
+    }
+
+    // load input SfMData scene
+    sfmData::SfMData sfmData;
+    if (!sfmDataIO::Load(sfmData, sfmDataFilename, sfmDataIO::ESfMData::ALL))
+    {
+        ALICEVISION_LOG_ERROR("The input SfMData file '" + sfmDataFilename + "' cannot be read.");
+        return EXIT_FAILURE;
+    }
+ 
+    // get imageDescriber type
+    const std::vector<feature::EImageDescriberType> describerTypes = feature::EImageDescriberType_stringToEnums(describerTypesName);
+
+    // features reading
+    feature::FeaturesPerView featuresPerView;
+    if (!sfm::loadFeaturesPerView(featuresPerView, sfmData, featuresFolders, describerTypes))
+    {
+        ALICEVISION_LOG_ERROR("Invalid features.");
+        return EXIT_FAILURE;
+    }
+
+    // matches reading
+    matching::PairwiseMatches pairwiseMatches;
+    if (!sfm::loadPairwiseMatches(pairwiseMatches, sfmData, matchesFolders, describerTypes, maxNbMatches, minNbMatches, useOnlyMatchesFromInputFolder))
+    {
+        ALICEVISION_LOG_ERROR("Unable to load matches.");
+        return EXIT_FAILURE;
+    }
+
+    if (extraInfoFolder.empty())
+        extraInfoFolder = fs::path(outputSfM).parent_path().string();
+
+    if (!fs::exists(extraInfoFolder))
+        fs::create_directory(extraInfoFolder);
+
+    // traingulate!
+    aliceVision::system::Timer timer;
+
+    if (sfmParams.minNbObservationsForTriangulation < 2)
+    {
+        // allows to use to the old triangulatation algorithm (using 2 views only) during resection.
+        sfmParams.minNbObservationsForTriangulation = 0;
+        // ALICEVISION_LOG_ERROR("The value associated to the argument '--minNbObservationsForTriangulation' must be >= 2 ");
+        // return EXIT_FAILURE;
+    }
+
+    //instantiate an sfmEngine for triangulation only
+    sfm::ReconstructionEngine_sequentialSfM sfmEngine(
```

Comment:
We should not use this reconstruction engine as it will perform a sequential SfM and we only want triangulation. Instead we should probably use one of the engine in `sfmTriangulation.hpp`